spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/test/AbstractSpannerIntegrationTest.java

```diff
@@ -0,0 +1,203 @@
+		}
+		catch (Exception e) {
+			this.setupFailed = true;
+			throw e;
+		}
+	}
+
+	protected void createDatabaseWithSchema() {
+		this.tableNameSuffix = String.valueOf(System.currentTimeMillis());
+		ConfigurableListableBeanFactory beanFactory = ((ConfigurableApplicationContext) this.applicationContext)
+				.getBeanFactory();
+		beanFactory.registerSingleton("tableNameSuffix", this.tableNameSuffix);
+		String instanceId = this.databaseId.getInstanceId().getInstance();
+		String database = this.databaseId.getDatabase();
+
+		if (!hasDatabaseDefined(instanceId, database)) {
+			this.databaseAdminClient.createDatabase(instanceId, database,
+					createSchemaStatements()).waitFor();
+			System.out.println(
+					this.getClass() + " - Integration database created with schema: " + createSchemaStatements());
+		}
+		else {
+			this.databaseAdminClient.updateDatabaseDdl(instanceId, database,
+					createSchemaStatements(), null).waitFor();
+			System.out.println(this.getClass() + " - schema created: " + createSchemaStatements());
+		}
+	}
+
+	protected List<String> createSchemaStatements() {
+		return Arrays.asList(Trade.createDDL(tableNameFor(Trade.class)));
+	}
+
+	protected Iterable<String> dropSchemaStatements() {
+		return Arrays.asList(Trade.dropDDL(tableNameFor(Trade.class)));
+	}
+
+	protected String tableNameFor(Class<Trade> type) {
+		return createDummyEntity(type).tableName();
+	}
+
+	protected <T> SpannerPersistentEntity createDummyEntity(Class<T> type) {
+		TypeInformation<T> typeInformation = mock(TypeInformation.class);
+		when(typeInformation.getType()).thenReturn(type);
+		SpannerPersistentEntity dummyTrade = new SpannerPersistentEntityImpl<>(typeInformation);
+		dummyTrade.setApplicationContext(this.applicationContext);
+		return dummyTrade;
+	}
+
+	@After
+	public void clean() {
+		try {
+			// this is to reduce duplicated errors reported by surefire plugin
+			if (this.setupFailed) {
+				return;
+			}
+			String instanceId = this.databaseId.getInstanceId().getInstance();
+			String database = this.databaseId.getDatabase();
+			if (hasDatabaseDefined(instanceId, database)) {
+				this.databaseAdminClient.updateDatabaseDdl(instanceId, database,
+						dropSchemaStatements(),
+						null).waitFor();
+				System.out.println("Integration database cleaned up!");
+			}
+		}
+		finally {
+			// we need to remove the extra bean created even if there is a failure at startup
+			DefaultListableBeanFactory beanFactory = (DefaultListableBeanFactory) this.applicationContext
+					.getAutowireCapableBeanFactory();
+			beanFactory.destroySingleton(TABLE_NAME_SUFFIX_BEAN_NAME);
+		}
+	}
+
+	private boolean hasDatabaseDefined(String instanceId, String database) {
+		Stream<String> databaseNames = StreamSupport
+				.stream(this.databaseAdminClient.listDatabases(instanceId).getValues().spliterator(), false)
+				.map(d -> d.getId().getDatabase());
+		return databaseNames.anyMatch(database::equals);
+	}
+
+	@Configuration
```

Comment:
A nit, but since we're explicitly pointing this configuration out in `@SpringBootTest(classes)`, we actually don't need this `@Configuration`, which is used for auto-detection of configuration.